OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -171,6 +172,15 @@ Action InitDisplayPanel(Widget panel)
 			windowModeDropdown.GetText = () => ds.Mode == WindowMode.Windowed ?
 				"Windowed" : ds.Mode == WindowMode.Fullscreen ? "Fullscreen (Legacy)" : "Fullscreen";
 
+			// Show Custom Resolution Input or Most Common Dropdown
+			var useCustomResolutionCheckbox = panel.Get<CheckboxWidget>("USE_CUSTOM_RESOLUTION_CHECKBOX");
+			var useCustomResolutionOnClick = useCustomResolutionCheckbox.OnClick;
+			useCustomResolutionCheckbox.IsVisible = () => ds.Mode == WindowMode.Windowed;
+			useCustomResolutionCheckbox.OnClick = () =>
+			{
+				useCustomResolutionOnClick();
```

Comment:
I don't think this is needed when you don't perform any extra options. (I.e. It should be save to remove line 177, and lines 179-182.)